spec/enumerables_spec.rb

```diff
@@ -0,0 +1,204 @@
+# rubocop:disable Layout/LineLength
+require_relative '../testing'
+
+RSpec.shared_context 'mainvariables' do
+  let(:words) { %w[dog door rod blade] }
+  let(:sentence) { %w[5 dog door rod blade] }
+  let(:ary) { [3, 3, 3, 2, 4, 2] }
+  let(:str) { %w[luigui mario boozer luffy zorro jinbei] }
+  let(:longest) { %w[cat sheep bear biggest] }
+end
+
+RSpec.describe 'testing my_each' do
+  include_context 'mainvariables'
+
+  it 'compare the my_each to ech when a block is given' do
+    expect([1, 2, 3, 4, 5].my_each(&proc { |n| p n.even? })).to eq([1, 2, 3, 4, 5].each(&proc { |n| p n.even? }))
```

Comment:
Please remove the the `p` statement, it's a bad practice to leave debugging statement in your codes